src/hooks/useReportIDs.tsx

```diff
@@ -130,7 +91,7 @@ function ReportIDsContextProvider({
     const derivedCurrentReportID = currentReportIDForTests ?? currentReportIDValue?.currentReportID;
     const {activeWorkspaceID} = useActiveWorkspace();
 
-    const policyMemberAccountIDs = getPolicyEmployeeListByIdWithoutCurrentUser(policies, activeWorkspaceID, accountID);
+    const policyMemberAccountIDs = useMemo(() => getPolicyEmployeeListByIdWithoutCurrentUser(policies, activeWorkspaceID, accountID), [policies, activeWorkspaceID, accountID]);
```

Comment:
> Now with the react-compiler, we do not have to add new useMemo right?

That's correct.

> @hurali97 was it a bottleneck in performance before? If so - can you try to remove this useMemo and see if performance degradation happens again?

@kirillzyusko Yes, `policyMemberAccountIDs` was changing during the render because of new instance being set each time, which causes `getOrderedReportByIDs` to be called in return. I will profile this PR with latest main in some time.

Reply:
@mountiny @kirillzyusko I profiled this PR on android with and without `useMemo` and the results are below.


Before:

We have only one entry of `ReportIDsContextProvider`.

![before-rc](https://github.com/Expensify/App/assets/47336142/3fe1611a-712c-4bbb-aaea-e891164b270a)


After:

We have two entries of `ReportIDsContextProvider` because `react-compiler` wasn't able to memoize `policyMemberAccountIDs`.

![after-rc](https://github.com/Expensify/App/assets/47336142/a8eef633-960a-4ee8-8b57-4d01f5e56861)


The reason could be that since we have passed an extra dependency `draftReports` to `getOrderedReportIDs` and a/c to react it breaks the rules. We need this dependency because we want to re-calculate `getOrderedReportIDs` whenever `draftReports` are updated.

Reply:
@kirillzyusko It makes sense and since this hook calls one of the most expensive functions of Expensify, it's better to use manual memoization here. Since `react-compiler` is in beta, we don't want to introduce any performance hits because of lack of memoization.